app/commands/v2/publish.rb

```diff
@@ -131,10 +130,10 @@ def publish_redirect_if_content_item_has_moved(new_location, previous_location,
         ) if draft_redirect
       end
 
-      def lookup_published_item(content_item)
+      def lookup_previous_item(content_item)
         ContentItemFilter.similar_to(
           content_item,
-          state: "published",
+          state: %w(published unpublished),
           base_path: nil,
           user_version: nil,
         ).first
```

Comment:
I wouldn't want to slow this PR up but I'd also prefer constraint over a noisy assertion here.